src/v/cluster/controller_backend.cc

```diff
@@ -1567,15 +1581,78 @@ ss::future<std::error_code> controller_backend::shutdown_on_current_shard(
     }
 }
 
+ss::future<> controller_backend::cleanup_orphan_files(
+  model::ntp ntp, model::revision_id rev) {
+    if (_self != 0) {
+        co_return;
+    }
+    auto topic_directory_path
+      = (std::filesystem::path(_data_directory) / ntp.topic_path());
+    ss::file topic_dir = co_await ss::open_directory(
+      topic_directory_path.string());
+    std::vector<ss::sstring> topic_files;
+    co_await topic_dir
+      .list_directory([&topic_files](ss::directory_entry entry) {
+          topic_files.push_back(entry.name);
+          return ss::make_ready_future<void>();
+      })
+      .done()
+      .finally([topic_dir]() mutable { return topic_dir.close(); });
+    for (const auto& file : topic_files) {
+        std::vector<ss::sstring> parts;
+        parts.reserve(2);
+        boost::algorithm::split(parts, file, boost::is_any_of("_"));
+        if (parts.size() != 2) {
+            continue;
+        }
+        int file_partition, file_revision;
+        try {
+            file_partition = std::stoi(parts[0]);
+            file_revision = std::stoi(parts[1]);
+        } catch (std::invalid_argument const& ex) {
+            continue;
+        }
```

Comment:
nit: it would be easier to read with regular expression and named groups

Reply:
fixed